torch/hub.py

```diff
@@ -105,12 +105,24 @@ def _get_torch_home():
 
 
 def _parse_repo_info(github):
-    branch = MASTER_BRANCH
     if ':' in github:
         repo_info, branch = github.split(':')
     else:
-        repo_info = github
+        repo_info, branch = github, None
     repo_owner, repo_name = repo_info.split('/')
+
+    if branch is None:
+        # The branch wasn't specified by the user, so we need to figure out the
+        # default branch: main or master. Our assumption is that if main exists
+        # then it's the default branch, otherwise it's master.
+        try:
+            urlopen(f"https://github.com/{repo_owner}/{repo_name}/tree/main/")
```

Comment:
nit: do we want to close the opened object? Maybe with a `with urlopen(...) as r:` block instead?